repobuilder/job.go

```diff
@@ -180,6 +180,28 @@ func NewRepoBuilderJob(opts JobOptions) (amboy.Job, error) {
 	return j, nil
 }
 
+// this function isolates our logic to alter a repoBuilderJob's
+// notaryKey for use when looking up signing keys
+func setNotaryKey(j *repoBuilderJob) {
+	if j.NotaryKey == "" {
+		j.NotaryKey = os.Getenv("NOTARY_KEY_NAME")
+		if j.NotaryKey == "" && j.release != nil {
+			if j.Distro.Type == DEB && (j.release.Series() == "3.0" || j.release.Series() == "2.6") {
+				j.NotaryKey = "richard"
+				j.NotaryToken = os.Getenv("NOTARY_TOKEN_DEB_LEGACY")
+			} else {
+				series := j.release.LTS()
+				if series == "" {
+					series = j.release.Series()
```

Comment:
For legacy versions, this will not return the next stable release series. For example, `3.7.1` would return `3.7` instead of `3.8`. I think that would be problematic.